### PR TITLE
Cmac/bfd 672 dme eob profile

### PR DESCRIPTION
**Pull request by mackec1**
### Change Details
Changes to achieve FIHR v4 conformance for the beneficiary DME element. This mostly involved making a new V4 DMEClaimTransformer class and some modifications to utility and test classes.

- gov.cms.bfd.server.war.r4.providers.DMEClaimTransformerV2.java (new)
- gov.cms.bfd.server.war.r4.providers.DMEClaimTransformerV2Test.java (new)
- minor mods to commins/carin
- minor mods to TransformerUtils.v2.java

### Acceptance Validation

This ticket is intended to meet FIHR v4 / CARIN conformance. Pursuant to that goal, some data decisions as noted in the previous section will also become part of the acceptance criteria.

The current version of the code does not meet the entire CARIN IG conformance; this is mostly due to a combination of our test data (i.e., bogus values that don't meet CARIN IG) or missing fields from the CCW data feed. Thus, while the current software meets much of the conformance, there are still errors that will be addressed based on ongoing remediation of data with both CARIN and CCW.

### Feedback Requested

FIHR v4 compliance suggestions.
Non-standard coding implemented.

### External References

https://jira.cms.gov/browse/BFD-672

### Security Implications

- [ ] new software dependencies

- [ ] altered security controls

- [x] new data stored or transmitted

FIHR v4, CARIN IG

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java [hunk 1]
```diff
@@ -163,7 +163,7 @@ public void transformSampleARecord() {
     Patient patient =
         BeneficiaryTransformerV2.transform(new MetricRegistry(), beneficiary, requestHeader);
 
-    System.out.println(fhirContext.newJsonParser().encodeResourceToString(patient));
+    // System.out.println(fhirContext.newJsonParser().encodeResourceToString(patient));
```
Comment (cbrunefearless) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java:166]: We should remove these lines if they are just used for debugging

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java [hunk 2]
```diff
@@ -70,6 +70,6 @@ static void assertMatches(CarrierClaim claim, ExplanationOfBenefit eob) throws F
 
     // TODO: finish tests based off V1
 
-    System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
+    // System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
```
Comment (cbrunefearless) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java:73]: same here

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HospiceClaimTransformerV2Test.java [hunk 3]
```diff
@@ -70,6 +70,6 @@ static void assertMatches(HospiceClaim claim, ExplanationOfBenefit eob) throws F
 
     // TODO: finish tests based off V1
 
-    System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
+    // System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
```
Comment (cbrunefearless) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HospiceClaimTransformerV2Test.java:73]: Same here

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/InpatientClaimTransformerV2Test.java [hunk 4]
```diff
@@ -87,6 +87,6 @@ static void assertMatches(InpatientClaim claim, ExplanationOfBenefit eob) throws
     // Test that the expected number of diagnoses are mapped
     Assert.assertEquals(9, eob.getDiagnosis().size());
 
-    System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
+    // System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
```
Comment (cbrunefearless) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/InpatientClaimTransformerV2Test.java:90]: Same here

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2Test.java [hunk 5]
```diff
@@ -74,6 +74,6 @@ static void assertMatches(OutpatientClaim claim, ExplanationOfBenefit eob) throw
     // TODO: Double check the assumed value
     Assert.assertEquals(5, eob.getDiagnosis().size());
 
-    System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
+    // System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
```
Comment (cbrunefearless) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2Test.java:77]: Same here

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2Test.java [hunk 6]
```diff
@@ -82,6 +82,6 @@ static void assertMatches(PartDEvent claim, ExplanationOfBenefit eob) throws FHI
 
     // TODO: finish tests based off V1
 
-    System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
+    // System.out.println(fhirContext.newJsonParser().encodeResourceToString(eob));
```
Comment (cbrunefearless) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2Test.java:85]: Same here

**Review by cbrunefearless — CHANGES_REQUESTED** [review]

**Review by cbrunefearless — APPROVED** [review]
lgtm

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/DMEClaimTransformerV2.java [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+    TransformerUtilsV2.addTotal(
+        eob,
+        TransformerUtilsV2.createTotalAdjudicationAmountSlice(
+            eob,
+            CcwCodebookVariable.CLM_TOT_CHRG_AMT,
+            C4BBAdjudication.PRIOR_PAYER_PAID,
+            claimGroup.getPrimaryPayerPaidAmount()));
+
+    // NCH_WKLY_PROC_DT => ExplanationOfBenefit.supportinginfo.timingDate
+    TransformerUtilsV2.addInformationWithDate(
```
Comment (Dranith) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/DMEClaimTransformerV2.java:103]: Just a note.  I changed how this is mapped on https://github.com/CMSgov/beneficiary-fhir-data/pull/488

We will want to make sure this merged first, and I'll update this under my PR.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/DMEClaimTransformerV2.java [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+
+      // REV_CNTR_PRVDR_PMT_AMT => ExplanationOfBenefit.item.adjudication
+      TransformerUtilsV2.addAdjudication(
+          item,
+          TransformerUtilsV2.createAdjudicationAmtSlice(
+              CcwCodebookVariable.REV_CNTR_PRVDR_PMT_AMT,
+              C4BBAdjudication.PAID_TO_PROVIDER,
+              line.getProviderPaymentAmount()));
+
+      // TODO - check w/jack if this is right ELIGIBLE
```
Comment (Dranith) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/DMEClaimTransformerV2.java:238]: Gut feeling tells me this is the correct slice.

**Review by Dranith — APPROVED** [review]
Looks good.

**Pull request merged**